src/translator/byte_array_util.cpp

```diff
@@ -45,14 +45,13 @@ MemoryGift getBinaryModelFromConfig(marian::Ptr<marian::Options> options) {
 }
 
 MemoryGift getBinaryShortlistFromConfig(marian::Ptr<marian::Options> options){
-    std::vector<std::string> vals = options->get<std::vector<std::string>>("shortlist");
-    if (vals.empty()){
-        std::cerr << "No path to shortlist file given" << std::endl;
-        std::exit(1);
-    }
-    std::string filename = vals[0];
-    return loadFileToMemory(filename,false);
+  std::vector<std::string> shortlist = options->get<std::vector<std::string>>("shortlist");
+  if (shortlist.empty()) {
+    std::cerr << "No path to shortlist file is given." << std::endl;
+    std::exit(1);
```

Comment:
Do we need to if-else this? `options->get<std::vector<std::string>>` will abort during runtime when a missing key is attempted to access with informative error messages, so this if else looks a bit redundant.

Reply:
It could be the case that the user specifies the key, but not the value. (Users always break things.)